ReactionHandler.js

```diff
@@ -64,6 +68,59 @@ class ReactionHandler {
     if (type === `ADD`) Member.roles.add(Role);
     if (type === `REMOVE`) Member.roles.remove(Role);
   }
+
+  async handleStarReaction(Reaction, User) {
+    await Reaction.fetch();
+    let starboardChannelId;
+    let isAlreadyOnStarboard = false;
+
+    if (Reaction.count === starConfig.starThreshold
+          && Reaction.message.author.bot === false
+          && Reaction.message.type === `DEFAULT`
+          && !starConfig.ignoreChannels.includes(Reaction.message.channel.name)) {
+      // Message meets the star threshold, is a non-bot, non-system message, and not in a black-listed channel
+      
+      // Construct the starboard message
+      const starMessage = new Discord.MessageEmbed()
+          .setColor(`#d74894`)
+          .setAuthor(`${Reaction.message.author.username}, your message belongs on the starboard:`, 
+                     Reaction.message.author.displayAvatarURL({format:`png`})) // Utilizing png, iPhone App seems to not handle webp
+          .setDescription(Reaction.message.content !== `` ? `${Reaction.message.content}` : `The message did not contain text, I've tried to attach it below.`)
+          .addField(`You're a ⭐ !`, `[Original message](https://discord.com/channels/${Reaction.message.guild.id}/${Reaction.message.channel.id}/${Reaction.message.id})`)
+          .setImage(Reaction.message.attachments.size === 0 ? `` : Reaction.message.attachments.values().next().value.url)
+          .setTimestamp(Reaction.message.createdTimestamp)
+          .setFooter(Reaction.message.id, Reaction.client.user.displayAvatarURL({format:`png`})) // Utilizing png, iPhone App seems to not handle webp
+
+      // Check if it should be posted on an over 18 starboard
+      if (Reaction.message.channel.parent != null 
+            && Reaction.message.channel.parent.name === starConfig.over18SectionName) {
+        starboardChannelId = starConfig.starboardOver18Id;
+      }
+      else {
+        starboardChannelId = starConfig.starboardId;
+      }
+
+      // Check if this message is already on starboard 
+      // (handle cases when star removed and re-added, i.e., trolling)
+      isAlreadyOnStarboard = await Reaction.client.channels.fetch(starboardChannelId)
+        .then(channel => channel.messages.fetch())
+        .then(messages => messages.some(m => m.embeds.length > 0 && m.embeds[0].footer.text === Reaction.message.id))
+        // Only run the check against messages that have embeds, i.e., posted by the bot
+        .catch(console.error);
+      
+      if(isAlreadyOnStarboard) {
+        // Discard, log potential abuse
+        console.log(`${User.username} re-met the threshold, perhaps stars were removed once at the threshold`);
+      }
+      else {
+        Reaction.client.channels.fetch(starboardChannelId)
+        .then(channel => channel.send(starMessage))
+        .catch(console.error);
```

Comment:
Should we have the bot reply reply to the original poster of the message in the channel it was posted in to notifying them it has been posted on the starboard? I think it would be nice.

Reply:
Went ahead and did this since it's possible the message meets the star threshold after many other message replies have come through. If it feels spammy, we can remove.